src/context.rs

```diff
@@ -363,6 +368,19 @@ impl<'a> Context<'a> {
             })
     }
 
+    pub fn get_jj_repo(&self) -> Option<&str> {
+        self.jj_repo
+            .get_or_init(|| {
+                let out = self.exec_cmd("jj", &["root", "--ignore-working-copy"])?;
```

Comment:
I think it might be better to do `begin_ancestor_scan` first for better performance.